### PR TITLE
Dark theme of meshery

### PR DESCRIPTION
**Pull request by Harshit-Dandriyal**
**Description**

This PR fixes #

**Notes for Reviewers**

Screenshot from 2022-11-14 15-44-27

Here is a list of already found existing issues in meshery that I am currently working on, if you are reviewing this PR, Ignore these sections.
https://docs.google.com/document/d/1Q1txcF99Z4OzW7W-PUaZySSo94JYXUL5asmCY9RWM58/edit?usp=sharing

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
No updates this past week, @Harshit-Dandriyal?

**Comment by Harshit-Dandriyal** [comment]
> No updates this past week, @Harshit-Dandriyal?

I am updating it time to time.

**Comment by leecalcote** [comment]
conflicts...

**Comment by Harshit-Dandriyal** [comment]
Fixing some things here..

**Comment by leecalcote** [comment]
Is there a flag that we can use to enable this mode without exposing the UI control to users? @Harshit-Dandriyal @MarioArriaga92 ?

**Comment by Harshit-Dandriyal** [comment]
https://user-images.githubusercontent.com/56961488/205129163-c087f3a6-2117-450f-9b1b-2153ba2868a2.mp4

@leecalcote I placed it under popover, so, It cannot be exposed much

**Comment by leecalcote** [comment]
@Harshit-Dandriyal ok. +1 from me.

**Comment by leecalcote** [comment]
However, this is what I see when I build this PR: 

Screenshot 2022-12-01 at 12 25 10 PM

**Comment by leecalcote** [comment]
All service mesh logos seem to be missing in the menu.
Screenshot 2022-12-01 at 12 28 00 PM

**Comment by Harshit-Dandriyal** [comment]
> However, this is what I see when I build this PR:
> 
> Screenshot 2022-12-01 at 12 25 10 PM

changed..

**Comment by Harshit-Dandriyal** [comment]
> All service mesh logos seem to be missing in the menu. Screenshot 2022-12-01 at 12 28 00 PM

This should not happen, let me see :thinking:

**Comment by leecalcote** [comment]
Screenshot 2022-12-01 at 12 29 29 PM

**Comment by leecalcote** [comment]
Each of the Application icons (k8s, docker-compose, helm) need light version of their logos. While you're at it, confirm that all other logos have a light and dark mode, too.

**Comment by Harshit-Dandriyal** [comment]
> All service mesh logos seem to be missing in the menu. Screenshot 2022-12-01 at 12 28 00 PM

I think this is not because of this PR, I am fixing it though, but this is happening in the latest master branch also.

**Comment by Harshit-Dandriyal** [comment]
image
@Lee < charcoal buttons :smile:

**Review by leecalcote — CHANGES_REQUESTED** [review]
Please include a documentation update for the Meshery UI contributing page on how this dark mode works.

**Comment by leecalcote** [comment]
Merge conflict.

**Comment by Udit-takkar** [comment]
@Harshit-Dandriyal can we have a toggle instead?

**Comment by Harshit-Dandriyal** [comment]
> @Harshit-Dandriyal can we have a toggle instead?

Yup, There can be.. Great suggestion

**Comment by MarioArriaga92** [comment]
> 

@Harshit-Dandriyal what @leecalcote means is some kind of environment variable we can set to `false` so the `Dark Mode` option/checkbox isn't rendered at all

**Comment by leecalcote** [comment]
> @Harshit-Dandriyal what @leecalcote means is some kind of environment variable we can set to `false` so the `Dark Mode` option/checkbox isn't rendered at all

Right on. Yes, that's right. I will suggest this potentially simpler path, though: If the mode switcher is buried in the user menu, this is probably hidden enough (as a last menu item under "Logout").

**Comment by Harshit-Dandriyal** [comment]
https://docs.google.com/document/d/1Q1txcF99Z4OzW7W-PUaZySSo94JYXUL5asmCY9RWM58/edit?usp=sharing

**Comment by Abhishek-kumar09** [comment]
Please merge the issues doc into this doc: https://docs.google.com/document/d/1ZZz_FRfnyHWqaGfufPY188z88BploZnW0XT2k7Hnq1s/edit#heading=h.c5hnflbvp66z

and add it into PR description.

File: ui/components/LoadingComponents/Animations/AnimatedLightMeshery.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
+import React, { useState, useEffect } from "react";
+import "./animatedMeshery.css";
```
Comment (Chadha93) [inline ui/components/LoadingComponents/Animations/AnimatedLightMeshery.js:2]:
```suggestion
import "./animatedMeshery.module.css";
```

**Comment by Chadha93** [comment]
@Harshit-Dandriyal Dark mode looks awesome. :sunglasses: . I just tried it. 
I guess this one is missed. 
image

**Comment by Chadha93** [comment]
Snackbars also need to reference themes and change colors. Reference: https://notistack.com/features/customization
image

**Comment by Chadha93** [comment]
@Harshit-Dandriyal Integration and lint checks are failing - `MuiThemeProvider`  _is defined but never used_. https://github.com/meshery/meshery/actions/runs/3711142815/jobs/6291306432#step:5:532

**Review by Nikhil-Ladha — CHANGES_REQUESTED** [review]
Some changes still needed:
1. The performance run count color could be changed to "white" in dark mode.
2. There is a leftover "=" operator on the performance profile's page.
Screenshot from 2022-12-16 22-41-48
3. The field labels in Modal should be "white". Also, having the Modal header color as dark jungle green would be better. Same as that we have for the Meshery header.
4. There seems to be some issue with the icons under the "Lifecycle" section of the navbar. Check the attached video. I am not sure if this is limited to me or you (or someone else) would be able to replicate it.

https://user-images.githubusercontent.com/29438875/208156086-855ecad5-e886-4cfb-be90-6bd07d44882f.mp4
5. The "Dark Mode" text should be clickable and work as a toggle. It shouldn't bee mandatory to click the checkbox itself. In fact, that's the general practice for checkboxes and radio buttons.
6. The profile page fields seem to not have a background color defined causing issues in dark mode. Should be handled in Meshery Cloud itself.

**Comment by leecalcote** [comment]
@Harshit-Dandriyal the "do-not-merge" label is still present on this PR. 

There are three files with merge conflicts.

**Comment by leecalcote** [comment]
3 merge conflicts to resolve

**Comment by Chadha93** [comment]
@Harshit-Dandriyal A lint error to fix. `MuiThemeProvider`
image

File: ui/components/DashboardComponent.js [hunk 2]
```diff
@@ -723,6 +725,22 @@ class DashboardComponent extends React.Component {
...
+      MuiPaper : { root : { backgroundColor : '#363636' } },
+      MuiFormLabel : {
+        root : {
+          "&$focused" : {
+            color : "#00B39F",
+          },
+        }
+      },
+    }
+  })
```
Comment (Abhishek-kumar09) [inline ui/components/DashboardComponent.js:743]: Would you extend on this?

Comment (Abhishek-kumar09) [reply ui/components/DashboardComponent.js:743]: same for https://github.com/meshery/meshery/blob/ebd105eb5789527616d5811b3a70e4b19c71b41e/ui/components/DashboardComponent.js#L682 where it has been used by someone else. Is it required to be here? If yes, then why do we have global theme.

File: ui/components/Filters.js [hunk 3]
```diff
@@ -203,49 +198,49 @@ function MesheryFilters({ updateProgress, enqueueSnackbar, closeSnackbar, user,
...
-  })
+  // const getMuiTheme = () => createTheme({
+  //   overrides : {
+  //     MuiInput : {
+  //       underline : {
+  //         "&:hover:not(.Mui-disabled):before" : {
+  //           borderBottom : "2px solid #222"
+  //         },
+  //         "&:after" : {
+  //           borderBottom : "2px solid #222"
```
Comment (Abhishek-kumar09) [inline ui/components/Filters.js:209]: can be deeleted?

File: ui/components/MesheryPatterns.js [hunk 4]
```diff
@@ -279,62 +284,62 @@ function MesheryPatterns({
...
-        },
-      },
-    }
-  });
+  // const getMuiTheme = () => createTheme({
+  //   overrides : {
+  //     MuiCheckbox : {
+  //       colorPrimary : {
+  //         color : "#607d8b",
+  //         "&$checked" : {
```
Comment (Abhishek-kumar09) [inline ui/components/MesheryPatterns.js:292]: this too, if not used, please delete.

File: ui/components/NodeDetails.js [hunk 5]
```diff
@@ -1,60 +1,60 @@
...
-      }
-    }
-  })
+  // const getMuiTheme = () => createTheme({
+  //   shadows : ["none"],
+  //   overrides : {
+  //     MuiInput : {
+  //       underline : {
+  //         "&:hover:not(.Mui-disabled):before" : {
+  //           borderBottom : "2px solid #222"
```
Comment (Abhishek-kumar09) [inline ui/components/NodeDetails.js:15]: same here.

File: ui/public/static/img/traefik_mesh-light.svg [hunk 6]
```diff
@@ -1 +1,23 @@
...
\ No newline at end of file
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" viewBox="0 0 128 128">
```
Comment (Abhishek-kumar09) [inline ui/public/static/img/traefik_mesh-light.svg:1]: btw, traefik Mesh icon was there in the component_svg folder. Do we need it?

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]
Looks Good to Me! Some comments to address. @Harshit-Dandriyal

File: ui/public/static/img/traefik_mesh-light.svg (hunk 6 above)
Comment (Harshit-Dandriyal) [reply ui/public/static/img/traefik_mesh-light.svg:1]: Yes, It was being still being in navigator.

File: ui/components/NodeDetails.js (hunk 5 above)
Comment (Harshit-Dandriyal) [reply ui/components/NodeDetails.js:15]: done

File: ui/components/MesheryPatterns.js (hunk 4 above)
Comment (Harshit-Dandriyal) [reply ui/components/MesheryPatterns.js:292]: done

File: ui/components/Filters.js (hunk 3 above)
Comment (Harshit-Dandriyal) [reply ui/components/Filters.js:209]: done

File: ui/components/DashboardComponent.js (hunk 2 above)
Comment (Harshit-Dandriyal) [reply ui/components/DashboardComponent.js:743]: Yes, It is required, Actually global theme is controlling the theme of that dashboard datatable, It was getting colors defined in the global theme, that's why I used this.

Comment (Abhishek-kumar09) [reply ui/components/DashboardComponent.js:743]: In that case, please put the functions in a new file with the context/comment of where and why it is used. Add it in this folder: https://github.com/meshery/meshery/tree/master/ui/themes

File: ui/public/static/img/traefik_mesh-light.svg (hunk 6 above)
Comment (Abhishek-kumar09) [reply ui/public/static/img/traefik_mesh-light.svg:1]: Okay, it looks like you haven't added it, but it was already there. The cleanup could be done in next PR.

**Comment by leecalcote** [comment]
Another conflict cropped up

**Comment by Abhishek-kumar09** [comment]
@Harshit-Dandriyal Would you resolve the merge conflicts finally?

**Comment by leecalcote** [comment]
Merge conflicts...

**Review by Abhishek-kumar09 — APPROVED** [review]
Lets Merge 🚀

**Pull request merged**